File: src/SOP8/광양/EgovLogin/base-model/src/main/java/egovframework/base/model/account/Option.java
package egovframework.base.model.account;

import lombok.*;
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Option {
	public enum Fields { user_sn, optn_cl, optn_sclas, optn_indx, optn_value }

	private int user_sn;

	@NonNull
	@Builder.Default
	private String optn_cl = "";

	@NonNull
	@Builder.Default
	private String optn_sclas = "";

	private int optn_indx;

	private String optn_value;

	public static final String autoIncreaseField = null;
	public static final String tableName = "acc_optn";
	public static final String[] primaryKeys = { "user_sn", "optn_cl", "optn_sclas", "optn_indx" };

}

File: src/SOP8/Base/Base.Model/Account/Option.cs
using System;
using dnsDapperDBUtil.DataAccessLayer.IDAL;

namespace Base.Model.Account
{
	public class Option : Table
	{
		public enum Fields { user_sn, optn_cl, optn_sclas, optn_indx, optn_value };
		public enum WriteFields { user_sn, optn_cl, optn_sclas, optn_indx, optn_value };

		public int user_sn { get; set; }
		public string optn_cl { get; set; }
		public string optn_sclas { get; set; }
		public int optn_indx { get; set; }
		public string/* nullable */ optn_value { get; set; }

		public static string TableName { get { return "acc_optn"; } }

		public override string GetTableName()
		{
			return TableName;
		}

		public override string GetPrimaryCondition()
		{
			return string.Format("{0} = {1} and {2} = '{3}' and {4} = '{5}' and {6} = {7}", Fields.user_sn, user_sn, Fields.optn_cl, optn_cl, Fields.optn_sclas, optn_sclas, Fields.optn_indx, optn_indx);
		}

		public override Type GetFieldType()
		{
			return typeof(Fields);
		}

		public override Type GetWriteFieldType()
		{
			return typeof(WriteFields);
		}

		public void FromCopy(Option obj)
		{
			this.user_sn = obj.user_sn;
			this.optn_cl = obj.optn_cl;
			this.optn_sclas = obj.optn_sclas;
			this.optn_indx = obj.optn_indx;
			this.optn_value = obj.optn_value;
		}
	}
}
